lib/state/src/index.ts

```diff
@@ -37,30 +37,33 @@ export interface State {
 }
 
 export function readState<T extends keyof State>(stage: T): State[T] | null {
-  if (fs.existsSync(stateFile)) {
-    const readStateContent = JSON.parse(fs.readFileSync(stateFile, { encoding: 'utf-8' }))
-    try {
-      return readStateContent[stage]
-    } catch {
-      return null
+  if (fs.existsSync(stateDir)) {
+    const file = `${stateDir}/${stage}.json`
+    if (fs.existsSync(file)) {
+      const readStateContent = JSON.parse(fs.readFileSync(file, { encoding: 'utf-8' }))
+      try {
+        return readStateContent
+      } catch {
+        return null
+      }
     }
   }
   return null
 }
 
 export function writeState<T extends keyof State>(stage: T, value: Partial<State[T]>): State[T] | null {
-  if (fs.existsSync(stateFile)) {
-    const readStateContent = JSON.parse(fs.readFileSync(stateFile, { encoding: 'utf-8' }))
-    for (const [key, val] of Object.entries(value)) {
-      readStateContent[stage] ||= {}
-      readStateContent[stage][key] = val
-    }
-    fs.writeFileSync(stateFile, JSON.stringify(readStateContent, null, 2))
+  const file = `${stateDir}/${stage}.json`
+  let readStateContent
+  if (fs.existsSync(stateDir)) {
+    if (fs.existsSync(file)) {
+      readStateContent = JSON.parse(fs.readFileSync(file, { encoding: 'utf-8' }))
+      for (const [key, val] of Object.entries(value)) {
+        readStateContent[key] = val
+      }
```

Comment:
**suggestion**: this could be done with `Object.assign` now:

```suggestion
      readStateContent = Object.assign(
        JSON.parse(fs.readFileSync(file, { encoding: 'utf-8' })),
        value
      )
```